datafusion/common/src/tree_node.rs

```diff
@@ -532,7 +532,7 @@ impl<T> Transformed<T> {
     }
 }
 
-/// Transformation helper to process tree nodes that are siblings.
+/// Transformation helper to process sequence of iterable tree nodes that are siblings.
 pub trait TransformedIterator: Iterator {
     fn map_until_stop_and_collect<
```

Comment:
I think explaining the intended behavior of this method would have helped me previously. Something like this perhaps (similarly for the next function too)

```suggestion
    /// Apples `f` to each item in this iterator
    /// 
    /// Visits all items in the iterator unless
    /// `f` returns an error or `f` returns TreeNodeRecursion::stop.
    ///
    /// # Returns 
    /// Error if f returns an error 
    ///
    /// Ok(Transformed) such that: 
    /// 1. `transformed` is true if any return from `f` had transformed true
    /// 2. `data` from the last invocation of `f`
    /// 3. `tnr` from the last invocation of `f` or `Continue` if the iterator is empty
    fn map_until_stop_and_collect<
```